### PR TITLE
Handle pagination in GetStatus API

### PR DESCRIPTION
**Pull request by davidmrdavid**
Addresses: https://github.com/Azure/azure-functions-durable-js/issues/236

**Context:**
When the `getStatus` / `getStatusBy` APIs are called, their response sizes are capped at 100 results. If more data can be fetched, a continuation token is passed-in in the response header.  Then, to fetch that data, we need to perform another request with that continuation token in the request's header.  This is from: https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-http-api#response-2

**The problem:**
Previously, the Durable Functions for JavaScript `getStatusBy` API did not handle the case where more than 100 results were available.  So it forced users with many results to manually construct the HTTP request by themselves and to manage the continuation token that explicitly. 

**This PR**
This PR augments the `getStatus` / `getStatusBy` APIs to manage continuation tokens automatically, therefore free'ing users from the burden of managing HTTP pagination themselves.

**Follow-up tasks**
To track this item in the Durable Python SDK

### CODE REVIEW

File: src/durableorchestrationclient.ts [hunk 1]
```diff
@@ -870,7 +870,23 @@ export class DurableOrchestrationClient {
...
     }
 
-    private async getStatusInternal(options: GetStatusOptions): Promise<AxiosResponse> {
+    /**
+     * Internal method that gets the status of all orchestration instances that
+     * match the specified conditions. It handles pagination automatically by
+     * calling itself recursively until no more continuation tokens are found.
+     * @param options Return orchestration instances which were created
+     *  after this Date.
+     * @param [continuationToken] Continuation token corresponding to the
```
Comment (davidmrdavid) [inline src/durableorchestrationclient.ts:879]: Just FYI that the bracket-enclosed names denote optional parameters in JSDoc

File: src/durableorchestrationclient.ts [hunk 2]
```diff
@@ -958,7 +974,34 @@ export class DurableOrchestrationClient {
...
+        let axiosConfig = undefined;
+        if (continuationToken) {
+            axiosConfig = {
+                headers: {
+                    "x-ms-continuation-token": continuationToken,
+                },
+            };
+        }
+
+        // We call the getStatus endpoint and construct a promise callback to handle the recursion
```
Comment (ConnorMcMahon) [inline src/durableorchestrationclient.ts:987]: I _really_ don't like that we have to do this pagination logic. This is really a consequence of us having an indirect proxy to the IDurableClient object via the HTTP apis (even with local RPC). The HTTP APIs are a subset of the C# APIs, meaning we have to do workarounds like this on the worker side, and some of these workarounds (like this one), can have major perf implications.

That being said, I think we have to do this, at least in the short term. I think this points to us needing to have a more comprehensive line using the localRpcEndpoint to IDurableClient implementations (both Durable 1.x and Durable 2.x) before we can get rid of this workaround,  and unfortunately we can't wait for that in the meantime.

**Review by ConnorMcMahon — APPROVED** [review]
I think this is the best we can do for now, so this gets my reluctant approval.

File: test/unit/durableclient-spec.ts [hunk 3]
```diff
@@ -244,6 +244,49 @@ describe("Durable client RPC endpoint", () => {
...
+            // The getStatusBy() method should do a GET to http://127.0.0.1:17071/durabletask/instances/?createdTimeFrom=2020-01-01T00:00:00Z&createdTimeTo=2020-01-01T23:59:59Z&runtimeStatus=Pending,Running,Completed,Terminated,Failed
+            const expectedUrl = new URL(`${testRpcOrigin}/durabletask/instances/`);
+            const createdTimeFrom = "2020-01-01T00:00:00.000Z";
+            const createdTimeTo = "2020-01-01T23:59:59.000Z";
+            const runtimeStatus = "Pending,Running,Completed,Terminated,Failed";
+
+            const scopeWithTokenResponse = nock(expectedUrl.origin)
+                .get(expectedUrl.pathname)
+                .query({ createdTimeFrom, createdTimeTo, runtimeStatus })
+                .reply(200, [null, null], { "x-ms-continuation-token": "myToken" });
```
Comment (ConnorMcMahon) [inline test/unit/durableclient-spec.ts:261]: small nit: Could we at least fill these with dummy objects. Returning an array filled with `null` as a shortcut feels a bit dirty. Ideally we would fill it with actual JSON representations of the data, but I realize that is a lot of work.

Comment (davidmrdavid) [reply test/unit/durableclient-spec.ts]: Of course, I agree the `null` was a little dirty.
Just replaced it with proper JSONs

File: src/durableorchestrationclient.ts (hunk 2 above)
Comment (davidmrdavid) [reply src/durableorchestrationclient.ts:987]: I've created [this](https://github.com/Azure/azure-functions-durable-extension/issues/1693) ticket to continue this conversation. Thanks for bringing this up :)

**Pull request merged**